src/gap.c

```diff
@@ -1523,6 +1523,22 @@ Obj FuncLOAD_DYN (
     if ( info == 0 )
         ErrorQuit( "call to init function failed", 0L, 0L );
 
+    // info->type should not be larger than kernel version
+    if (info->type / 10 > GAP_KERNEL_API_VERSION)
+        ErrorMayQuit("LOAD_DYN: kernel module built for newer "
+                     "version of GAP",
+                     0L, 0L);
+
+    // info->type should not have an older major version
+    if (info->type / 10000 < GAP_KERNEL_API_VERSION / 1000)
```

Comment:
The right part could become `GAP_KERNEL_MAJOR_VERSION`, which would be a bit more readable perhaps?

Alternatively, have macros for extracting the minor/major part of a version. Then write
`if (MAJOR_VERSION(info->type/10) < MAJOR_VERSION(GAP_KERNEL_API_VERSION))`

Reply:
Have switched to using `GAP_KERNEL_MAJOR_VERSION`. Would be happy to introduce macros, but not sure if anyone would ever use them.